scm-step/src/main/java/org/jenkinsci/plugins/workflow/steps/scm/GitStep.java

```diff
@@ -43,6 +65,7 @@
 
     private final String url;
     private String branch = "master";
+    private StandardCredentials credentials;
```

Comment:
Why ? For standard component which get serialized to xml I know we can't do that, be a workflow step is serialized according to it's @DataBound attributes.